src/jdk.jshell/share/classes/jdk/jshell/execution/LocalExecutionControl.java

```diff
@@ -90,11 +94,24 @@ public void load(ClassBytecodes[] cbcs)
     private static byte[] instrument(byte[] classFile) {
         var cc = ClassFile.of();
         return cc.transformClass(cc.parse(classFile),
-                        ClassTransform.transformingMethodBodies((cob, coe) -> {
-                            if (coe instanceof BranchInstruction)
-                                cob.invokestatic(CD_Cancel, "stopCheck", ConstantDescs.MTD_void);
-                            cob.with(coe);
-                        }));
+                        ClassTransform.transformingMethodBodies(
+                            CodeTransform.ofStateful(() -> {
+                                HashSet<Label> prior = new HashSet<>();
+                                return (cob, coe) -> {
+                                    switch (coe) {
+                                    case BranchInstruction branch:
```

Comment:
Can we use switch with `->` instead of `:` to avoid the error-prone breaks?

Reply:
Hi Chen! Excellent suggestion, thanks. Updated in 30c5f1a3c8b.